### PR TITLE
new HMS binding

### PR DESCRIPTION
**Pull request by swatchy2dot0**
Hi openhab team,

this is my first own binding and I'd like to share it with the community. I'm totally new to git, so I hope my pull request is OK? What else do I have to do? What about a WIKI page and an config guide?

Thanks so far,
Tom

### CODE REVIEW

**Comment by teichsta** [comment]
Hi Tom,

thanks for this contribution, which looks quite good already.

> What else do I have to do?

wait for the Binding to be reviewed.

> What about a WIKI page and an config guide?

Meanwhile you could indeed start writing a Wiki page like available for all other Bindings. Please add a note to your page that the Binding will be available with the upcoming 1.7 release (see Nest Binding Page as an example).

To things that came to eyes immediately:

* please remove the generic comments on class level and add a brief description of what the specific classes are doing.
* please remove overridden methods which you are not enhancing somehow
* please change @since to 1.7.0 in all classes
* please change version in MANIFEST to 1.7.0.qualifier

More to come in the upcoming Review.

Thanks, Thomas E.-E.

File: bundles/binding/org.openhab.binding.hms/src/main/java/org/openhab/binding/hms/internal/HMSBinding.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+			cul = CULManager.getOpenCULHandler(deviceName, CULMode.SLOW_RF);
+			cul.registerListener(this);
+		} catch (CULDeviceException e) {
+			logger.error("Can't open cul device", e);
+			cul = null;
+		}
+	}
+
+	public void deactivate() {
+		logger.debug("Deactivating FS20 binding");
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hms/src/main/java/org/openhab/binding/hms/internal/HMSBinding.java]: copy page issue?

File: bundles/binding/org.openhab.binding.hms/src/main/java/org/openhab/binding/hms/internal/HMSBinding.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+			double temperature = Integer.parseInt(woHeader.substring(5, 6) + woHeader.substring(2, 4)) / 10.0;
+			boolean isNegative = (s1.charAt(0) & 8) == 8;
+			if (isNegative) {
+				temperature = -temperature;
+			}
+			double humidity = Integer.parseInt(woHeader.substring(6, 8) + woHeader.substring(4, 5)) / 10.0;
+
+			logger.info("device: " + device + ", T: " +(!isNegative?" ":"")+ temperature + ",\tH: " + humidity + ", Bat.: "
+					+ batteryStatus);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.hms/src/main/java/org/openhab/binding/hms/internal/HMSBinding.java]: use parameterised logging

**Comment by teichsta** [comment]
ok, i had a quick look into your binding and added some minor comments inline. Despite those the following comments still apply:

* remove the generic comments on class level and add a brief description of what the specific classes are doing.
* remove overridden methods which you are not enhancing somehow
* change @since to 1.7.0 in all classes
* change version in MANIFEST to 1.7.0.qualifier
* add your binding to bundles/binding/pom.xml
* add your binding to distribution/pom.xml

Would you be able to incorporate the feedback soon? Please also squash your commits into one again after you've incorporated my feedback.

Thanks again, Thomas E.-E.

**Comment by teichsta** [comment]
Any chance to get the feedback incorporated today?

**Comment by swatchy2dot0** [comment]
Sorry, I'm on the road. Answering you by today evening.

Thanks Tom
Am 23.05.2015 08:34 schrieb "Thomas Eichstädt-Engelen" <
notifications@github.com>:

**Comment by swatchy2dot0** [comment]
Hi Thomas,

I just committed the changes related to your comments. Is there something
more I've to do? Is there a need for another pull request?

Thanks,
Tom

By the way, you're German, aren't you? Me too ;-)

2015-05-20 23:45 GMT+02:00 Thomas Eichstädt-Engelen <
notifications@github.com>:

**Comment by teichsta** [comment]
please keep the according wiki page in mind. Once you've finished it please link to it here. Thanks, Thomas E.-E.

**Comment by swatchy2dot0** [comment]
Wiki page can be found here https://github.com/openhab/openhab/wiki/HMS-Binding

**Comment by teichsta** [comment]
thanks a lot!

- sent from a mobile device -

**Comment by timoschober** [comment]
Hi all,
i have done some changes on the HMS Binding, so that i can use fs20 and hms with the cul on the same time.
And i add the support for the HMS100WD (water alarm).

I like to checkin the changes to github. Is this possible?
How can I do this?

**Comment by swatchy2dot0** [comment]
Hi Timo,

creating the initial version of the HMS Binding was my first experience with github, that's why I'm not sure whether my following proposal is the best solution ;-)

Since HMS Binding is now part of openhab, just create a fork of the official repository (like described here: https://github.com/openhab/openhab/wiki/How-To-Contribute), commit your changes to this fork and create a pull request.

It is really nice to hear that my binding is used by other and even extended. What where that changed for a parallel usage of FS20 and HMS? I'm actually using HMS and FS20 binding in parallel, without changes.

Thanks,
Tom

**Comment by timoschober** [comment]
Hi Tom,
i used your code before it was part of openhab ;-)

I changed the function which get the Cul device:
cul = CULManager.getOpenCULHandler(deviceName, CULMode.SLOW_RF, properties);

i only copy the line from the fs20-binding and then it works parallel. I don't know how it works for you without this change. May be you use an other configuration of openhab.cfg

Thank you very much for your work!
Timo

**Pull request merged**